sse/src/main/scala/io/finch/sse/ServerSentEvent.scala

```diff
@@ -0,0 +1,51 @@
+package io.finch.sse
+
+import java.nio.charset.Charset
+
+import cats.Show
+import com.twitter.concurrent.AsyncStream
+import com.twitter.io.Buf
+import io.finch.{Encode, Ok, Output, Text}
+import io.finch.internal.{BufText, LowPriorityToResponseInstances}
+
+case class ServerSentEvent[A](
+  data: A,
+  id: Option[String] = None,
+  event: Option[String] = None,
+  retry: Option[Long] = None
+)
+
+object ServerSentEvent extends LowPriorityToResponseInstances {
+  type EventStream[A] = AsyncStream[ServerSentEvent[A]]
+
+  def stream[A](sse: EventStream[A]): Output[EventStream[A]] = {
+    Ok(sse).withHeader("Cache-Control" -> "no-cache").withHeader("Connection" -> "keep-alive")
+  }
+
+  implicit def sseAsyncToResponse[A](implicit
+     e: Encode.Aux[A, Text.EventStream]
+  ): Aux[AsyncStream[A], Text.EventStream] =
+    asyncResponseBuilder((a, cs) => e(a, cs).concat(Buf.Utf8("\n")))
+
+  implicit def encodeEventStream[A](implicit s: Show[A]): Encode.Aux[ServerSentEvent[A], Text.EventStream] = {
+    Encode.instance[ServerSentEvent[A], Text.EventStream]({ (event: ServerSentEvent[A], c: Charset) =>
+      encodeEvent[A](event, c, { (a: A, cs: Charset) => BufText(s.show(a), cs) })
+    })
+  }
+
+  def encodeEvent[A](sse: ServerSentEvent[A], cs: Charset, e: (A, Charset) => Buf): Buf = {
```

Comment:
I left this available with the thought that it can be used by consumers of the library to construct their own server sent events. But I can make this private.